qiskit/transpiler/passes/unroller.py

```diff
@@ -42,8 +46,27 @@ def run(self, dag):
             current_node = dag.multi_graph.node[node]
             if current_node["op"].name in self.basis:  # If already a base, ignore.
                 continue
-            decomposition_rules = current_node["op"].decompositions()
+
             # TODO: allow choosing other possible decompositions
+            try:
```

Comment:
wouldn't be better to check `is not None`?

Reply:
I've changed this block to try to get the `decomposition_rules` and except only a `RecursionError`.